core/services/chainlink/config.go

```diff
@@ -180,3 +181,24 @@ func (s *Secrets) setEnv() error {
 	}
 	return nil
 }
+
+func (c *Config) DetectDuplicates(f *Config) (err error) {
```

Comment:
Do you think that duplicates will remain the only case? Or would it make sense call this something more general, like `PreValidateConfig()`? :shrug:

Reply:
Good question!

I like the naming of `PreValidateConfig` 👍 

The way that the `setFrom` hierarchy currently works is that it de-duplicates and de-nils any configuration values before passing on the config object to the `node` subcommands. So to generalize this method, I think we would need to see a path forward with asserting nil checks in `PreValidateConfig()` in addition to the uniqueness checks asserted with this change. 

However, the value for a field left unset in the passed in configuration (i.e. `-c /.../config.toml -c /.../config1.toml`) will be unset when it hits the`node` subcommands as is. Then, subcommand validation can be used to assert that a needed field is set before overlaying overrides and the fallback. i.e. I don't think nil checks in `PreValidateConfig` would add any value, leaving just the uniqueness (or duplication detection) checks.

Reply:
nil checks don't make sense before normal validation, since we actually expect many nils in each sub-config.

I can't think of any other checks that are helpful at this point :thinking: